src/devices/chihiro/JvsIo.cpp

```diff
@@ -97,203 +118,246 @@ int JvsIo::Jvs_Command_GetBoardId()
+
+	{ // Input capabilities
+		ResponseBuffer.push_back(0x01); // Player button-sets
+		ResponseBuffer.push_back(JVS_MAX_PLAYERS); // number of players
+		ResponseBuffer.push_back(13); // 13 button switches per player
+		ResponseBuffer.push_back(0);
+
+		ResponseBuffer.push_back(0x02); // Coin slots
+		ResponseBuffer.push_back(JVS_MAX_COINS); // number of coin slots
+		ResponseBuffer.push_back(0);
+		ResponseBuffer.push_back(0);
+
+		ResponseBuffer.push_back(0x03); // Analog inputs
+		ResponseBuffer.push_back(JVS_MAX_ANALOG); // number of analog input channels
+		ResponseBuffer.push_back(16); // 16 bits per analog input channel
+		ResponseBuffer.push_back(0);
+
+		// TODO : Rotary input (0x04), JVS_MAX_ROTARY, 0, 0
+
+		// TODO : Keycode input (0x05), 0, 0, 0
+
+		// TODO : Screen pointer input (0x06), Xbits, Ybits, JVS_MAX_POINTERS
+
+		// TODO : Switch input (0x07), 0, 0, 0
+	}
+
+	{ // Output capabilities
+		// TODO : Card system (0x10), JVS_MAX_CARDS, 0, 0
+
+		// TODO : Medal hopper (0x11), max?, 0, 0
+
+		ResponseBuffer.push_back(0x12); // General-purpose outputs
+		ResponseBuffer.push_back(6); // number of outputs
+		ResponseBuffer.push_back(0);
+		ResponseBuffer.push_back(0);
+
+		// TODO : Analog output (0x13), channels, 0, 0
+
+		// TODO : Character output (0x14), width, height, type
+
+		// TODO : Backup data (0x15), 0, 0, 0
+	}
+
+	ResponseBuffer.push_back(0x00); // End of capabilities
 
 	return 0;
 }
 
-int JvsIo::Jvs_Command_ReadSwitchInputs(uint8_t* data)
+int JvsIo::Jvs_Command_20_ReadSwitchInputs(uint8_t* data)
 {
-	uint8_t players = data[1];
-	uint8_t bytesPerPlayer = data[2];
+	static jvs_switch_player_inputs_t default_switch_player_input;
+	uint8_t nr_switch_players = data[1];
+	uint8_t bytesPerSwitchPlayerInput = data[2];
 
-	ResponseBuffer.push_back(0x01);
+	ResponseBuffer.push_back(ReportCode::Handled);
 
 	ResponseBuffer.push_back(Inputs.switches.system);
 
-	for (int i = 0; i < players; i++) {
-		for (int j = 0; j < bytesPerPlayer; j++) {
+	for (int i = 0; i < nr_switch_players; i++) {
+		for (int j = 0; j < bytesPerSwitchPlayerInput; j++) {
+			// If a title asks for more switch player inputs than we support, pad with dummy data
+			jvs_switch_player_inputs_t &switch_player_input = (i >= JVS_MAX_COINS) ? default_switch_player_input : Inputs.switches.player[i];
 			uint8_t value
-				= (i >= 2) ? 0 // We only support two players, so we pad any extras with null bytes
-				: (j == 0) ? Inputs.switches.player[i].GetByte0()
-				: (j == 1) ? Inputs.switches.player[i].GetByte1()
+				= (j == 0) ? switch_player_input.GetByte0()
+				: (j == 1) ? switch_player_input.GetByte1()
 				: 0; // Pad any remaining bytes with 0, as we don't have that many inputs available
 			ResponseBuffer.push_back(value);
 		}
 	}
-		
+
 	return 2;
 }
 
-int JvsIo::Jvs_Command_ReadCoinInputs(uint8_t* data)
+int JvsIo::Jvs_Command_21_ReadCoinInputs(uint8_t* data)
 {
-	uint8_t slots = data[1];
+	static jvs_coin_slots_t default_coin_slot;

```

Comment:
why this static?

Reply:
To have simpler code below

Reply:
... by reading out-of-bound indices from the static structs, so that the defaults defined in each of those structs won't have to be repeated here.